### ISSUE TITLE
Applying LDS/FDS to classic machine learning models

### ISSUE THREAD

**Issue by luopx** [YyzHarry/imbalanced-regression#31]
Hi! This work is really fantastic! However, I found it hard to apply LDS/FDS to classic machine learning models like random forest. For example, after getting the effective label density with LDS, how should I use this?

**Comment by YyzHarry** [comment]
Hi, thanks for your interest. For LDS and FDS, they should be easy to apply to classic machine learning models.

For LDS, since it operates on the label space, it is independent of the model you use. For example, if using random forest (regressor), referring to the `fit()` method of the [sklearn RandomForestRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html#sklearn.ensemble.RandomForestRegressor.fit) (and other classic methods like SVM), there is an argument `sample_weight`, where you can provide the weight of each input sample. You can then use the effective label density by LDS to compute the weights.

As for FDS, it might be less straightforward; but since it only needs feature statistics, if the algorithm can generate intermediate features for each input, it can also be adapted, depending on the actual method you use.

**Comment by luopx** [comment]
> 

Thank you very much for your reply! This really helped me a lot!
